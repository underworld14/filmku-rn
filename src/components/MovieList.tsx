import { TouchableOpacity, Image, Text, View, TouchableOpacityProps } from "react-native";
import AntDesignIcon from "@expo/vector-icons/AntDesign";
import GenreChip from "components/GenreChip";
import { getGenresByIds } from "utils/helpers";

type Props = TouchableOpacityProps & {
  poster_path?: string;
  title?: string;
  vote_average?: number;
  vote_count?: number;
  genre_ids?: number[];
};

export default function MovieList({
  poster_path,
  title,
  vote_average,
  vote_count,
  genre_ids,
  ...props
}: Props) {
  return (
    <TouchableOpacity className="flex-row w-full mb-4" {...props}>
      <Image className="w-24 h-32" source={{ uri: `https://image.tmdb.org/t/p/w500${poster_path}` }} />
      <View className="ml-4 w-3/4">
        <Text className="font-text-bold text-base">{title}</Text>
        <Text className="font-text-regular text-xs mt-2 text-gray-400">
          <AntDesignIcon name="star" color="#FFC319" /> {vote_average}/{vote_count}
        </Text>
        <View className="flex-row mt-3 flex-wrap w-4/5">
          {getGenresByIds(genre_ids || []).map((genre, index) => (
            <GenreChip key={index} name={genre.name} />
          ))}
        </View>
      </View>
    </TouchableOpacity>
  );
}
