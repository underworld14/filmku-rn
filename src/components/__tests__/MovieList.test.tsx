import MovieList from "components/MovieList";
import renderer from "react-test-renderer";

describe("MovieList", () => {
  it("renders correctly", () => {
    const tree: any = renderer.create(<MovieList />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
