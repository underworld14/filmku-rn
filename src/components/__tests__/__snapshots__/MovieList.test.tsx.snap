// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "marginBottom": 16,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://image.tmdb.org/t/p/w500undefined",
      }
    }
    style={
      Array [
        Object {
          "width": 96,
        },
        Object {
          "height": 128,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "marginLeft": 16,
        },
        Object {
          "width": "75%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish_700Bold",
          },
          Object {
            "fontSize": 16,
            "lineHeight": 24,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Mulish_400Regular",
          },
          Object {
            "fontSize": 12,
            "lineHeight": 16,
          },
          Object {
            "marginTop": 8,
          },
          Object {
            "color": "#9ca3af",
          },
        ]
      }
    >
      <Text />
       
      /
    </Text>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "marginTop": 12,
          },
          Object {
            "flexWrap": "wrap",
          },
          Object {
            "width": "80%",
          },
        ]
      }
    />
  </View>
</View>
`;
